pkg/retry/retry.go

```diff
@@ -81,3 +83,31 @@ func WithBackoff(
 		}
 	}
 }
+
+// Opinionated considers timeouts, temporary, DNS and connection refused errors as retryable.
+func Opinionated(err error) bool {
+	if errors.Is(err, syscall.ECONNREFUSED) || errors.Is(err, syscall.EAGAIN) {
+		return true
+	}
+
+	var dnsError *net.DNSError
+	if errors.As(err, &dnsError) {
+		return true
+	}
+
+	type temporary interface {
+		Temporary() bool
+	}
+	if t := temporary(nil); errors.As(err, &t) {
+		return t.Temporary()
+	}
+
+	type timeout interface {
+		Timeout() bool
+	}
+	if t := timeout(nil); errors.As(err, &t) {
+		return t.Timeout()
+	}
```

Comment:
I'd consistently define variable before or in the if here. (Hint: you can also do `var temporary interface { ... }`, then you don't even need to explicitly do `temporary(nil)`.